### PR TITLE
VAULT-1303 when a request to vault fails, show namespace if set

### PR DESCRIPTION
**Pull request by hghaf099**

### CODE REVIEW

**Comment by hghaf099** [comment]
I have also ran the new test I added in vault-ent to test this new functionality and it passes.

File: api/response.go [hunk 1]
```diff
@@ -41,12 +41,14 @@ func (r *Response) Error() error {
 
 	r.Body.Close()
 	r.Body = ioutil.NopCloser(bodyBuf)
+	ns := r.Header.Get("X-Vault-Namespace")
```
Comment (ncabatoff) [inline api/response.go:44]: Can you use consts.NamespaceHeaderName instead of the literal string?

**Review by ncabatoff — APPROVED** [review]
Minor change suggested, looks good overall, nice work!

File: api/response.go (hunk 1 above)
Comment (hghaf099) [reply api/response.go:44]: Done

**Pull request merged**